internal/server/handler.go

```diff
@@ -128,6 +133,13 @@ func (s *Server) getHost(bucket, endPoint string) string {
 	return bucket + "." + endPoint
 }
 
+func (s *Server) isUsingIAMRoleCredentials() bool {
+	if s.cfg.S3Settings.AccessKeyID == "" && s.cfg.S3Settings.SecretAccessKey == "" {
+		return true
+	}
+	return false
```

Comment:
```suggestion
	return s.cfg.S3Settings.AccessKeyID == "" && s.cfg.S3Settings.SecretAccessKey == ""
```

nit: this can be simplified as shown